### PR TITLE
Remove tbb link when not used

### PR DESCRIPTION
**Pull request by vanclevstik**
**What does this PR do? Please describe:**

Trying to build the library from source on ARM64 CPU, https://github.com/facebookresearch/fairseq2/pull/4 has added support for tbb being optional, but I had to remove one more line to get cmake to work.

I think this should still work without problems on Intel where TBB is supported because of this line https://github.com/facebookresearch/fairseq2/blob/main/fairseq2n/src/fairseq2n/CMakeLists.txt#L119 but worth double checking if my thinking is right.

Related to #1

**Does your PR introduce any breaking changes? If yes, please list them:**
List of all backwards-incompatible changes.

**Check list:**
- [ ] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [x] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**?
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

**Comment by cbalioglu** [comment]
Thanks for the fix @vanclevstik! I also realized that bug after merging the PR and was planning to fix it in the upcoming v0.1.1 (hope to release tomorrow). Merging your PR though to unblock.

**Review by cbalioglu — APPROVED** [review]

**Comment by vanclevstik** [comment]
Thanks for merging @cbalioglu ! If you need any more help with testing/fixing on ARM64 devices do let me know, happy to help :)

**Comment by cbalioglu** [comment]
You are welcome! :) I finally got an M2 machine the other day and I think with today's PR (#13) ARM64 builds are now in good shape. My final battle is with Windows/WSL2. I was expecting to finish v0.1.1 work by yesterday, so things got delayed a bit, but hope to wrap everything up before the end of the week.

**Pull request merged**
